templates/tools/generate-metadata/main.rs

```diff
@@ -5,9 +5,8 @@ extern "Rust" {
 }
 
 fn main() -> Result<(), std::io::Error> {
-	let ink_project = unsafe { __ink_generate_metadata() };
-	let contents = serde_json::to_string_pretty(&ink_project)?;
-	std::fs::create_dir("target").ok();
-	std::fs::write("target/metadata.json", contents)?;
+	let metadata = unsafe { __ink_generate_metadata() };
+	let contents = serde_json::to_string_pretty(&metadata)?;
+	print!("{}", contents);
```

Comment:
Interesting trick to go through the `stdout` bytes here. Why was this design chosen?

Reply:
Previously when we only had `ink!` metadata, the generated `metadata-gen` package wrote directly to  `target/metadata.json`. 

Now we require combining the `ink!` metadata with the extended contract metadata. We need some way of marshalling the data returned by invoking `__ink_generate_metadata` back to the calling `cargo-contract` process where the metadata can be combined.

Note that in the first iteration of this, the extended metadata generated by `cargo-contract` and the `ink!` contract metadata were combined inside the `ink_metadata` crate, so the data was marshalled in the other direction (via code generation). However as discussed previously this is considerably less flexible in terms of compatibility with different versions of `ink!`.

Using json serialized to `stdout` is a more idiomatic way of inter process communication, and provides us with that property of backwards/forwards compatibility with `ink!` versions.